src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -416,6 +416,27 @@ public static IntPtr AllocateTypeAssociatedMemory(Type type, int size)
                 }
             }
         }
+
+        [LibraryImport(QCall, EntryPoint = "ReflectionInvocation_SizeOf")]
+        [SuppressGCTransition]
+        private static partial int SizeOf(QCallTypeHandle handle);
+
+        public static unsafe int SizeOf(RuntimeTypeHandle type)
+        {
+            if (type.IsNullHandle())
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.type);
+
+            TypeHandle handle = type.GetNativeTypeHandle();
+            if (!handle.IsTypeDesc && handle.AsMethodTable()->IsGenericTypeDefinition)
```

Comment:
Would it be better to do this check on the unmanaged side? It is tightly coupled with the unmanaged implementation of TypeHandle::GetSize and how it behaves in corner cases.

Reply:
I can move it to the unmanaged side if you'd prefer. I don't have a preference.